get.go

```diff
@@ -237,3 +250,60 @@ func getChallengeResponse(c *config, ec *attest.EncryptedCredential, aikBytes []
 		Secret: secret,
 	}, nil
 }
+
+func checkMDNSDomain(originalURL string, headers *http.Header) (string, error) {
+	fmt.Println("checking mdns domain")
+	parsedURL, err := url.Parse(originalURL)
+	if err != nil {
+		return originalURL, fmt.Errorf("parsing the mdns url: %w", err)
+	}
+
+	host := parsedURL.Host
+	if !strings.HasSuffix(host, ".local") {
+		return originalURL, nil
+	}
+
+	mdnsIP, mdnsPort := discoverMDNS(host)
+	if mdnsIP == "" { // no reply
+		return originalURL, nil
+	}
+
+	headers.Add("Host", parsedURL.Host)
+	newURL := strings.ReplaceAll(originalURL, host, mdnsIP)
+	// Remove any port in the original url
+	if port := parsedURL.Port(); port != "" {
+		newURL = strings.ReplaceAll(newURL, port, "")
+	}
+
+	// Add any possible port from the mdns response
+	if mdnsPort != "" {
+		newURL = strings.ReplaceAll(newURL, mdnsIP, fmt.Sprintf("%s:%s", mdnsIP, mdnsPort))
+	}
+
+	return newURL, nil
+}
+
+func discoverMDNS(host string) (string, string) {
+	// Make a channel for results and start listening
+	entriesCh := make(chan *mdns.ServiceEntry, 4)
+	defer close(entriesCh)
+
+	// Start the lookup.
+	// The channel is buffered so it doesn't block.
+	// The Lookup here has its own timeout until it receives a response.
+	// We use a select with a timeout to read because we don't know if we didn't
+	// get the response yet or if there will be no response at all.
+	mdns.Lookup(MDNSServiceName, entriesCh)
```

Comment:
this `Lookup` is calling `log.Printf` in some places. That output escapes the pluggable plugin that puts all the rest of the output in the json response, not sure why. On the other hand it doesn't seem to break the caller that tries to parse json. Maybe it's sent to stderr ? I'm not sure.

Reply:
yes `2> /dev/null` makes it disappear. So it's stderr.